jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1413,6 +1413,15 @@ public String changeSessionId()
         return session.getId();
     }
 
+    protected void onRequestLog()
+    {
+        // Don't allow pulling more parameters
+        _contentParamsExtracted = true;
```

Comment:
Can we split this PR into two.
This is the fix that we need to fix the actual hard bug.  The not mutating the response is a nice to have feature.  I'm currently working on some ideas for that in jetty-12, so once those work out, it would be good to step in the same direction.

Reply:
To me that means ...

1. The restoring of committed metadata
2. The isMutable checks in Response

That what you mean?

Reply:
My understanding of the actual bug we are trying to fix is to stop implementations of requestLog from blocking trying to read parameters.   So the `_contentParamsExtracted = true` change here is the actual fix.
Making sure that the response headers and status are immutable after commit is a second order issue.

Reply:
At a minimum the headers and status code need to be copied into place. (what `restoreCommittedMetadata` does)

Reply:
Basically, restore the committed state is step 1.

Step 2 is to prevent the altering of state after commit.

Reply:
No.   Let's say that I don't really care about if we log incorrect data if the headers/status are changed after commit ( I do a bit, but let's pretend).     Even if I don't care about that, I still care a lot if the 3rd party request logger tries to get a parameter and ends up blocking in extractParameters.    So let's fix that hard bug and then deal with the immutability of committed headers for request logs in a second PR.

Reply:
Opened PR #7183 for the first part.

Reply:
Not going to open second PR, as I don't like PRs depending on other PRs (the merge gets messy, and it's impossible to not rebase force-push in those scenarios).
Lets get #7183 merged, then we can base part 2 on that from `jetty-10.0.x` (post-merge)